mrp_unbuild_rebuild_variant/models/mrp_unbuild_rebuild_variant.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, exceptions, fields, models
+
+
+class MrpUnbuildRebuildVariant(models.Model):
+
+    _name = "mrp.unbuild.rebuild.variant"
+
+    name = fields.Char(
+        default=lambda self: self.env["ir.sequence"].next_by_code(self._name),
+        readonly=True,
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("done", "Done")],
+        required=True,
+        default="draft",
+        readonly=True,
+    )
+    # TODO: WHAT SHOULD BE COMPUTED HERE
+    unbuild_product_id = fields.Many2one("product.product")
+    unbuild_bom_id = fields.Many2one("mrp.bom", compute="_compute_unbuild_product_id")
+    unbuild_location_id = fields.Many2one("stock.location")
```

Comment:
IMO we need only one location for both unbuild and rebuild, with a default value being the stock of the company's warehouse. Moreover the field should be displayed in the view for users having group multiple locations enabled.